azkaban-exec-server/src/main/java/azkaban/execapp/action/KillExecutionAction.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2012 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.execapp.action;
+
+import azkaban.ServiceProvider;
+import azkaban.execapp.FlowRunnerManager;
+import azkaban.trigger.builtin.SlaChecker;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+
+import azkaban.trigger.TriggerAction;
+
+public class KillExecutionAction implements TriggerAction {
+
+  public static final String type = "KillExecutionAction";
+
+  private static final Logger logger = Logger
+      .getLogger(KillExecutionAction.class);
+
+  private String actionId;
+  private int execId;
+
+  public KillExecutionAction(String actionId, int execId) {
+    this.execId = execId;
+    this.actionId = actionId;
+  }
+
+  @Override
+  public String getId() {
+    return actionId;
+  }
+
+  @Override
+  public String getType() {
+    return type;
+  }
+
+  @SuppressWarnings("unchecked")
+  public static KillExecutionAction createFromJson(Object obj) {
+    return createFromJson((HashMap<String, Object>) obj);
+  }
+
+  public static KillExecutionAction createFromJson(HashMap<String, Object> obj) {
+    Map<String, Object> jsonObj = (HashMap<String, Object>) obj;
+    String objType = (String) jsonObj.get("type");
+    if (!objType.equals(type)) {
```

Comment:
consider it as part of refactoring https://github.com/azkaban/azkaban/issues/1098